.travis.yml

```diff
@@ -14,38 +13,50 @@ addons:
-
 cache:
   apt: true
   directories:
     - $HOME/.stack/
     - .stack-work/
 
-before_install:
-  - mkdir -p $HOME/texlive && curl -L https://github.com/urdh/texlive-blob/releases/download/20160619/texlive.tar.xz | tar -JxC $HOME/texlive
-  - export PATH="$PATH:$HOME/texlive/bin/x86_64-linux"
-  # Install stack
-  - mkdir -p $HOME/.local/bin
-  - export PATH=$HOME/.local/bin:$PATH
-  - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C $HOME/.local/bin '*/stack'
-  # Install node 6; from https://github.com/nodejs/nan/blob/master/.travis.yml
-  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install 7
-  - npm install npm
-  - mv node_modules npm
-  - npm/.bin/npm --version
-  - npm/.bin/npm install
-  - sudo pip install tox
-  - sudo add-apt-repository -y ppa:openjdk-r/ppa
-  - sudo apt-get -qq update
-  - sudo apt-get install -y openjdk-8-jdk --no-install-recommends
-  - sudo update-java-alternatives -s java-1.8.0-openjdk-amd64
-
 matrix:
   include:
     - env:
-        - TRAVIS_TARGET=default
+        - TRAVIS_TARGET=python
+      language: python
+      python: 3.7-dev
+      before_install:
+        - sudo apt-get -qq update
+        - sudo apt-get install python2.7
+        - sudo apt-get install python3.5
+        - sudo pip install tox
+      script: make test-python
+    - env:
+        - TRAVIS_TARGET=default JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+      before_install:
+        - mkdir -p $HOME/texlive && curl -L https://github.com/urdh/texlive-blob/releases/download/20160619/texlive.tar.xz | tar -JxC $HOME/texlive
+        - export PATH="$PATH:$HOME/texlive/bin/x86_64-linux"
+        # Install stack
+        - mkdir -p $HOME/.local/bin
+        - export PATH=$HOME/.local/bin:$PATH
+        - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C $HOME/.local/bin '*/stack'
+        # Install node 6; from https://github.com/nodejs/nan/blob/master/.travis.yml
+        - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install 7
+        - npm install npm
+        - mv node_modules npm
+        - npm/.bin/npm --version
+        - npm/.bin/npm install
+        - sudo add-apt-repository -y ppa:openjdk-r/ppa
+        - sudo apt-get -qq update
+        - sudo apt-get install -y openjdk-8-jdk --no-install-recommends
+        - sudo update-java-alternatives -s java-1.8.0-openjdk-amd64
+        # - tlmgr update --self
+        # - tlmgr install standalone #standalone.sty
+        # - tlmgr install cmbright #cmbright font
+        - sudo pip install --upgrade Sphinx
+        - npm install -g mocha
+        - npm install
+      script: make test-c test-haskell test-java test-javascript
```

Comment:
Does this line mean we no longer test the generator on travis? 
What should be our plan to make sure the generator isn't broken long term?

What python version is supposed to be use for "make python" or "make c" or "make all"?

Reply:
The generator isn't tested on travis but if I'm not mistaken, it wasn't tested before either. The previous command in the default job/target was `make test`, essentially leading to the same sequence as here + test-python (which is now a separate job). A couple of unit test *files* exist for the generator under `generator/tests/` but they are empty. AFAIK, generating the bindings has been done by manually calling the generator.

I'll clarify the python version documentation for the generator commands a bit.

Reply:
great! Sorry I missed that the command was make test before as well.  Once long ago we did a make all on travis.